portalocker/utils.py

```diff
@@ -281,13 +281,18 @@ def try_close():  # pragma: no cover
                 if fail_when_locked:
                     try_close()
                     raise exceptions.AlreadyLocked(exception) from exc
+            except Exception as exc:
+                # Something went wrong with the locking mechanism.
+                # Wrap in a LockException and re-raise:
+                try_close()
+                raise exceptions.LockException(exc) from exc
 
-                # Wait a bit
+            # Wait a bit
 
         if exception:
             try_close()
             # We got a timeout... reraising
-            raise exceptions.LockException(exception)
+            raise exception
```

Comment:
Similar to the previous one, these functions should always return some form of `LockException` for existing scenarios. Other types of exceptions might still come through, but those weren't caught previously so that won't influence backwards compatibility.